cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/NewTwoWindingsTransformerConversion.java

```diff
@@ -0,0 +1,426 @@
+            .setR(convertedT2xModel.r)
+            .setX(convertedT2xModel.x)
+            .setG(convertedT2xModel.end1.g + convertedT2xModel.end2.g)
+            .setB(convertedT2xModel.end1.b + convertedT2xModel.end2.b)
+            .setRatedU1(convertedT2xModel.end1.ratedU)
+            .setRatedU2(convertedT2xModel.end2.ratedU);
+        identify(adder);
+        connect(adder);
+        TwoWindingsTransformer tx = adder.add();
+        convertedTerminals(tx.getTerminal1(), tx.getTerminal2());
+
+        setToIidmRatioTapChanger(convertedT2xModel, tx);
+        setToIidmPhaseTapChanger(convertedT2xModel, tx);
+
+        setToIidmPhaseAngleClock(convertedT2xModel, tx);
+        setRegulatingControlContext(convertedT2xModel, tx);
+    }
+
+    private void setToIidmRatioTapChanger(ConvertedT2xModel convertedT2xModel, TwoWindingsTransformer tx) {
+        TapChangerConversion rtc = convertedT2xModel.end1.ratioTapChanger;
+        if (rtc == null) {
+            return;
+        }
+
+        RatioTapChangerAdder rtca = newRatioTapChanger(tx);
+        setToIidmRatioTapChanger(rtc, rtca);
+    }
+
+    private void setToIidmPhaseTapChanger(ConvertedT2xModel convertedT2xModel, TwoWindingsTransformer tx) {
+        TapChangerConversion ptc = convertedT2xModel.end1.phaseTapChanger;
+        if (ptc == null) {
+            return;
+        }
+
+        PhaseTapChangerAdder ptca = newPhaseTapChanger(tx);
+        setToIidmPhaseTapChanger(ptc, ptca);
+    }
+
+    protected RatioTapChangerAdder newRatioTapChanger(TwoWindingsTransformer tx) {
+        return tx.newRatioTapChanger();
+    }
+
+    protected PhaseTapChangerAdder newPhaseTapChanger(TwoWindingsTransformer tx) {
+        return tx.newPhaseTapChanger();
+    }
+
+    private void setToIidmPhaseAngleClock(ConvertedT2xModel convertedT2xModel, TwoWindingsTransformer tx) {
+        // add phaseAngleClock as an extension, cgmes does not allow pac at end1
+        if (convertedT2xModel.end1.phaseAngleClock != 0) {
+            String reason = "Unsupported modelling: twoWindingsTransformer with phaseAngleClock at end1";
+            ignored("phaseAngleClock end1", reason);
+        }
+        if (convertedT2xModel.end2.phaseAngleClock != 0) {
+            TwoWindingsTransformerPhaseAngleClock phaseAngleClock = new TwoWindingsTransformerPhaseAngleClock(tx,
+                convertedT2xModel.end2.phaseAngleClock);
+            tx.addExtension(TwoWindingsTransformerPhaseAngleClock.class, phaseAngleClock);
+        }
+    }
+
+    private void setRegulatingControlContext(ConvertedT2xModel convertedT2xModel, TwoWindingsTransformer tx) {
+        CgmesRegulatingControlRatio rcRtc = setContextRegulatingDataRatio(convertedT2xModel.end1.ratioTapChanger);
+        CgmesRegulatingControlPhase rcPtc = setContextRegulatingDataPhase(convertedT2xModel.end1.phaseTapChanger);
+
+        context.regulatingControlMapping().forTransformers().add(tx.getId(), rcRtc, rcPtc);
+    }
```

Comment:
Shouldn't it be more logical directly in `RegulatingControlMappingForTransformers`?

Reply:
I think should be here as we did in generators and the rest of controlled devices. This method establishes the link between the IIDM transformer and the controlID that will be used later to define the regulatingControl.